client/a8c-for-agencies/sections/sites/sites-dashboard/index.tsx

```diff
@@ -138,26 +143,39 @@ export default function SitesDashboard() {
 	}, [ isLoading, isError, sitesViewState.filters, filtersMap ] );*/
 
 	useEffect( () => {
+		// If the favorites filter is set, make sure to update the filter and correctly add the is_favorite param to URLs.
+		filter.showOnlyFavorites = isFavoriteFilter;
+		const favoritesParam = isFavoriteFilter ? '?is_favorite' : '';
```

Comment:
This is just a note for us. We should move this logic to a separate function and write unit tests for it, given that the resulting URL depends on multiple variables and the outcome is not easy to predict. Writing tests will also document the expected behavior, which a reader from the future will certainly appreciate.

Reply:
I have added a new issue to our board to track the necessary refactor.

Reply:
Creating unit tests for this function is not a high priority. We can address it after the launch as a maintenance issue.

Thank you for asking, @coder-karen .